File: backend/app/schemas/assistant.py
import uuid

from pydantic import BaseModel, Field


class AssistantMessage(BaseModel):
    role: str = Field(pattern="^(user|assistant)$")
    content: str


class AssistantChatRequest(BaseModel):
    message: str = Field(min_length=1, max_length=2000)
    history: list[AssistantMessage] = Field(default_factory=list, max_length=20)


class AssistantChatResponse(BaseModel):
    reply: str
    referenced_product_ids: list[uuid.UUID]

File: frontend/src/api/assistant.ts
import { apiClient } from "./client";

export interface AssistantMessage {
  role: "user" | "assistant";
  content: string;
}

export interface AssistantChatResponse {
  reply: string;
  referenced_product_ids: string[];
}

export async function askAssistant(
  message: string,
  history: AssistantMessage[],
): Promise<AssistantChatResponse> {
  const { data } = await apiClient.post<AssistantChatResponse>("/assistant/chat", {
    message,
    history,
  });
  return data;
}
